docs/source/impl-guide/normalization.rst

```diff
@@ -3,50 +3,105 @@
 Normalization
 !!!!!!!!!!!!!
 
-Certain insertion or deletion alleles may be represented ambiguously
-when using conventional sequence normalization, resulting in
-significant challenges when comparing such alleles.
+In VRS, "normalization" refers to the process of converting a given
+variation representation into a canonincal form.  Left- and
+right-shuffling are examples of normalization for sequence variants.
+VRS extends this concept to apply to all classes of VRS Variation.
+VRS normalization minimizes a class of ambiguity that impedes
+comparison of variation across systems.
 
-VRS describes a "fully-justified" normalization algorithm
-inspired by NCBI's Variant Overprecision Correction Algorithm [1]_.
+Implementations MUST provide a normalize function that accepts any
+Variation object and returns a normalized Variation.  Guidelines for
+these functions are below.
+
+
+General Normalization Rules
+@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+* VRS normalization functions are homomorphic: That is, the input and
+  output objects are always of the same `type`.
+* Object types that do not have VRS normalization rules are returned
+  as-is.  That is, all types of Variation MUST be supported, even if
+  such objects are unchanged.
+* VRS normalization functions are idempotent: Normalizing a normalized
+  Variation object returns an object with the same data.
```

Comment:
I was avoiding phrases like "the same object" because I believe that implementations should actually return a copy.

Or, more specifically, the function should always operate in-place or on a copy, irrespective of internal conditions.

Reply:
Fair point on "copy". I was trying to emphasize that a normalized object won't be changed (e.g., attribute order scrambled), as that would break normalization.